### PR TITLE
New documentation site with Docusaurus

### LINKED ISSUE #344: [Bug]: Fix and update doc deployment workflow
### Contact Details

_No response_

### What happened?

The following worklfow builds documentation that has not run for over a year.
https://github.com/Green-Software-Foundation/carbon-aware-sdk/actions/workflows/pages/pages-build-deployment

Do we need to fix so the latest gets deployed in a format that may be issue to consume by others?

Other projects in GSF use [Docasaurus](https://docusaurus.io/) to build their documentation, an example such project is : https://github.com/Green-Software-Foundation/sci-guide , currently deployed to : https://sci-guide.greensoftware.foundation/

Best solution to this issue would involve - restructuring the current [docs](/docs) to be buildable and deployable as a Docasaurus project on github pages.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### LINKED ISSUE #382: [Bug]: Some GHA were failed due to empty workflow
### Contact Details

_No response_

### What happened?

Following actions were failed because workflow file is empty (all lines are commented out). Is it temporally?

* .github/workflows/run-sdkCLI-githubaction.yml
* .github/workflows/2.a-deploy.yaml
* .github/workflows/run-sdkCLI-githubaction.yml

### client

WebAPI (Default)

### Relevant log output

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by danuw**
# Pull Request

Issue Number: #344

## Summary

New documentation site using docusaurus currently available at https://green-software-foundation.github.io/carbon-aware-sdk/

## Changes

- fixes #344 
- fixes #382 - workflows have not been renamed with .ignore extension (they were completely commented out before and that was not enough)
- documentation moved from `/docs` to `/casdk-docs/docs`
- new pipeline dedicated to building the documentation site
  - deploys on changes to dev affecting documentation folders
  - moves the samples folder at deploy time to ensure it stays in the repo's root folder though deploys in the documentation
- includes disclaimer in banner for non graduated projects (addresses #416)
- includes link to carbon hack in banner
- includes blog with new pages for each release

## Checklist

- [x] Local Tests Passing?
- [x] CICD and Pipeline Tests Passing?
- [x] Documentation Updates Made?
- [x] This is not a breaking change. If it is, please describe it below.

## Is this a breaking change?

The documentation now sites elsewhere (in `/casdk-docs/docs`)

## Anything else?

Other comments, collaborators, etc.

> Please follow
> [GitHub's suggested syntax](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword)
> to link Pull Requests to Issues via keywords

This PR Closes #344 #416 #382

### CODE REVIEW

**Comment by Sophietn** [comment]
@danuw Docusaurus looks great! Thank you for all the work getting it to this point

File: casdk-docs/package.json [hunk 1]
```diff
@@ -0,0 +1,46 @@
+{
+  "name": "casdk-docs",
+  "version": "0.0.0",
+  "private": true,
+  "scripts": {
+    "docusaurus": "docusaurus",
+    "start": "docusaurus start",
```
Comment (Willmish) [inline casdk-docs/package.json:7]: Possibly can edit docusaurs scripts to add the required copy command for copying samples docs: `cp -r ./samples/ ./casdk-docs/docs/ && <OTHER_COMMANDS>`

Comment (danuw) [reply casdk-docs/package.json:7]: Cool will give it a go soon. Thanks

**Comment by vaughanknight** [comment]
Doing a check now

**Review by vaughanknight — COMMENTED** [review]
Approved!

**Comment by vaughanknight** [comment]
> Approved!

I hadn't clicked approve after reviewing in another tab :)

**Pull request merged**